src/observe-tag.ts

```diff
@@ -35,102 +51,108 @@ const parentAndChildrenElements = (node, tagName) => {
  *
  * The mutation-observer will unobserve any reactions here when the node is removed.
  */
-module.exports = tagFunction => {
-	let tagResult
+export default (tagFunction) => {
+	let tagResult;
 	const buildAndReplaceTag = () => {
 		// if there is an existing tagResult, it is the last rendering, and so we want to re-render over it
-		let oldTag = tagResult
-		let removedElementWithFocusData = {}
+		let oldTag = tagResult;
+		let removedElementWithFocusData: RemovedElementDataStore = {};
 
 		// remove oldTag first so that we unobserve before we re-observe
 		if (oldTag) {
 			// we need to blow away any old focus data we had
-			removedElementWithFocusData = {}
+			removedElementWithFocusData = {};
 
 			// determine if this element (or any element under it) had focus
-			const oldTagHasFocusedElement = oldTag.contains(document.activeElement)
+			const oldTagHasFocusedElement = oldTag.contains(document.activeElement);
 
 			// if an element had focus, copy over all the selection data (so we can copy it back later)
 			if (oldTagHasFocusedElement) {
+				// we'll assume that the element is an HTMLInputElement, in reality other kinds of elements will be caught here,
+				// but that's fine, since they have null as selection attributes, and setting them to null is fine
+				const activeElement = document.activeElement as HTMLInputElement;
```

Comment:
Pretty much what it says in the comment. This is because Typescript (and somewhat inaccurately so) complains that "Element" doesn't have the selection attributes (e.g. `selectionStart`), in reality, these will be null, which I guess for Typescript is just as bad, but for us, that is fine.